app/src/main/java/org/desarrolladorslp/teckersapp/service/RetrofitManager.kt

```diff
@@ -0,0 +1,69 @@
+package org.desarrolladorslp.teckersapp.service
+
+import android.content.Context
+import android.widget.Toast
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.google.android.material.snackbar.Snackbar
+import kotlinx.android.synthetic.main.activity_main.*
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Response
+import org.desarrolladorslp.teckersapp.BuildConfig
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import java.io.IOException
+
+class RetrofitManager {
+
+    companion object {
+
+        private var authInstance: Retrofit? = null
+        private var baseUrl= BuildConfig.BaseUrl
+        private lateinit var applicationLayout :ConstraintLayout
+
+        fun setActivity(constraintLayout: ConstraintLayout)
+        {
+            applicationLayout = constraintLayout
+        }
+
+        fun instance(): Retrofit? {
+
+            if (authInstance == null) {
+
+                val client = OkHttpClient.Builder()
+                    .addInterceptor(object : AppCredentialsInterceptor() {
+                        @Throws(IOException::class)
+                        override fun intercept(chain: Interceptor.Chain): Response {
+                            var request = chain.request()
+                            var response = chain.proceed(request)
+                            if(response.code() == 500)
```

Comment:
This code is already written on `APIEndpoint`, please move it to a new interceptor so you write this just once.